### PR TITLE
Build command should not allow runtime options

### LINKED ISSUE #9618: Build command should only accept built-time options
### Description

The build command should not allow persisting runtime options. The reasons are:

* Not aligned with best practices where sensitive data is persisted into the server image/container. E.g.: password, urls, etc.
* Provides a more clear separation and more clarity for users in regards to the different configuration stages

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #9618

### CODE REVIEW

File: quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java [hunk 1]
```diff
@@ -62,6 +60,7 @@
 import io.quarkus.deployment.builditem.LaunchModeBuildItem;
 import io.quarkus.deployment.builditem.RuntimeConfigSetupCompleteBuildItem;
 import io.quarkus.deployment.builditem.StaticInitConfigSourceProviderBuildItem;
+import io.quarkus.dev.console.QuarkusConsole;
```
Comment (DGuhr) [inline quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java:63]: unused, pls remove.

**Review by DGuhr — CHANGES_REQUESTED** [review]
trying your PR out locally atm. Seems all feature options do not get picked up anymore.

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java [hunk 2]
```diff
@@ -53,4 +54,11 @@ void failIfDevProfile(LaunchResult result) {
                 () -> "The Error Output:\n" + result.getErrorOutput() + "doesn't contains the expected string.");
         assertEquals(4, result.getErrorStream().size());
     }
+
+    @Test
+    @Launch({ "build", "--db=postgres", "--db-username=myuser", "--db-password=mypassword", "--http-enabled=true" })
+    void testFailRuntimeOptions(LaunchResult result) {
+        CLIResult cliResult = (CLIResult) result;
+        cliResult.assertError("Unknown options: '--db-username=myuser', '--db-password=mypassword', '--http-enabled=true'");
```
Comment (DGuhr) [inline quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java:62]: I think we should also mask the password here. Could again leak into logs when ruunning it with wrong config in a ci build due to missing knowledge. now the value, which was taken from a secure place, is in those logs. so please re-add the masking

Comment (pedroigor) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java:62]: Makes sense but we don't really sanitize this specific output when printing out invalid options.

We should fix this in a separate PR as a follow to https://github.com/keycloak/keycloak/pull/8944. We missed this output when doing that PR.

We should also figure out what to do when you wrongly type the name of a "masked" option, which should lead to the same problem.

That said, I would enhance this in a separate issue.

**Comment by pedroigor** [comment]
@DGuhr Fixed the features support and added the test we were missing. We should now cover this area.

**Comment by DGuhr** [comment]
@pedroigor will re-check

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java (hunk 2 above)
Comment (pedroigor) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java:62]: One thing here though. The value is not yet "taken from a secure place" but the raw value passed to the CLI. That message happens in an early stage, prior to resolving values from config sources.

Comment (DGuhr) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java:62]: You are right. Created https://github.com/keycloak/keycloak/issues/9720 - not sure we can/should really do something about mistyped keys, though, at least not without exceptionally adding complexity to check for blurriness in keys ;)

**Review by DGuhr — DISMISSED** [review]

**Review by DGuhr — DISMISSED** [review]

**Review by DGuhr — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]
Approving per @DGuhr 's approval

**Pull request merged**